Storage/Bookmarks/Bookmarks.swift

```diff
@@ -14,6 +14,10 @@ public protocol SearchableBookmarks: class {
     func bookmarksByURL(_ url: URL) -> Deferred<Maybe<Cursor<BookmarkItem>>>
 }
 
+public protocol CountableBookmarks: class {
+    func numberOfBookmarks() -> Deferred<Maybe<Int>>
```

Comment:
@rnewman What are your thoughts on exposing a method like this for bookmarks/history?